background/handle-site-access.js

```diff
@@ -0,0 +1,24 @@
+chrome.permissions.contains(
+  {
+    permissions: ["webRequest", "webRequestBlocking", "cookies"],
+    origins: ["https://scratch.mit.edu/*", "https://api.scratch.mit.edu/*", "https://clouddata.scratch.mit.edu/*"],
+  },
+  async (hasPermissions) => {
+    if (hasPermissions) {
+      // Load remaining scripts
+      await import("./handle-fetch.js");
+      await import("./handle-cookie.js");
+      await import("./handle-auth.js");
+      await import("./handle-messages.js");
+
+      await import("./get-userscripts.js");
+      await import("./get-persistent-scripts.js");
```

Comment:
I do not see the benefit of moving these scrips here. Why not let them have errors?